src/sql/workbench/services/notebook/common/notebookServiceImpl.ts

```diff
@@ -132,6 +149,30 @@ export class NotebookService extends Disposable implements INotebookService {
 		}
 	}
 
+	private hookNotebookThemesAndConfigListener(): void {
+		if(this._configurationService) {
+			this.updateNotebookThemes();
+			this._register(this._configurationService.onDidChangeConfiguration(e => {
+				if (e.affectsConfiguration(OVERRIDE_EDITOR_THEMING_SETTING)) {
```

Comment:
This ensures we recalculate our theme logic based on the configuration setting change.